### PR TITLE
feat: make wandb-core the default SDK backend

### PR DESCRIPTION
**Pull request by kptkin**
Description
-----------

What does the PR do? Include a concise description of the PR contents.

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

**Review by dmitryduev — APPROVED** [review]
🎉 🔥

File: wandb/sdk/wandb_require.py [hunk 1]
```diff
@@ -46,22 +41,11 @@ def require_service(self) -> None:
...
-        if is_require_legacy_service():
-            raise UnsupportedError(
-                'Cannot use wandb.require("core") because legacy-service'
-                " is already required--did you use"
-                ' wandb.require("legacy-service") or set the'
-                " WANDB__REQUIRE_LEGACY_SERVICE environment variable?"
-            )
-        os.environ[_REQUIRE_CORE] = "true"
+        wandb.termwarn(
+            "This is currently the default behavior and does not need to be set."
```
Comment (timoffex) [inline wandb/sdk/wandb_require.py:45]: The user won't know where this message is coming from, so it should be more explicit.

> `wandb.require("core")` is redundant as it is now the default behavior.

Comment (kptkin) [reply wandb/sdk/wandb_require.py:45]: yep i was just looking at the wording here and i didn't like it, i'll update to your wording, thanks

File: wandb/sdk/lib/_settings_toposort_generated.py [hunk 2]
```diff
@@ -213,6 +213,8 @@
     "_notebook",
     "disabled",
     "_offline",
+    "_require_legacy_service",
+    "_require_core",
```
Comment (kptkin) [inline wandb/sdk/lib/_settings_toposort_generated.py:217]: @dmitryduev i think there is a bug in the toposort script

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,12 @@ Please add to the relevant subsections under Unreleased below on every PR where
 
 - Detect the notebook name in VS Code's built-in jupyter server (@dmitryduev in https://github.com/wandb/wandb/pull/8311)
 
+### Changed
+
+- The `wandb` package now uses the `wandb-core` binary as the default runtime service. To revert to the legacy behavior,
```
Comment (timoffex) [inline CHANGELOG.md:24]: "`wandb-core` binary" feels like an implementation detail. We've referred to this as "core" 
and "wandb-core" before in the changelog and communications, so how about

> The new "core" backend, previously activated using `wandb.require("core")`, is now used by default. [the rest]

File: wandb/cli/cli.py
Comment (timoffex) [inline wandb/cli/cli.py:1]: Will this use `wandb-core` for other CLI commands by default?

File: wandb/sdk/wandb_run.py [hunk 4]
```diff
@@ -4260,13 +4260,13 @@ def _footer_notify_wandb_core(
...
         """Prints a message advertising the upcoming core release."""
-        if quiet or settings._require_core:
+        if quiet or not settings._require_legacy_service:
             return
 
         printer.display(
-            "The new W&B backend becomes opt-out in version 0.18.0;"
-            ' try it out with `wandb.require("core")`!'
-            " See https://wandb.me/wandb-core for more information.",
+            "The legacy backend is deprecated. In version 0.19.0, `wandb-core` will become "
```
Comment (timoffex) [inline wandb/sdk/wandb_run.py:4267]: Are we sure we want to commit to 0.19 for that purpose? How about "In future versions" instead?

Comment (kptkin) [reply wandb/sdk/wandb_run.py:4267]: yeah i was thinking about it, let's not commit, good point

File: wandb/cli/cli.py
Comment (kptkin) [reply wandb/cli/cli.py:1]: most shouldn't being using any service, but good point let's verify this

**Pull request merged**